src/EffectiveCSharp.Analyzers/Common/DiagnosticIds.cs

```diff
@@ -15,4 +15,8 @@ internal static class DiagnosticIds
     internal const string MinimizeBoxingUnboxing = "ECS0900";
     internal const string BeAwareOfValueTypeCopyInReferenceTypes = "ECS0901";
     internal const string UseSpanInstead = "ECS1000";
+    internal const string PreferDeclarationInitializersToAssignmentStatement = "ECS1200";
+    internal const string PreferDeclarationInitializersExceptNullOrZero = "ECS1201";
```

Comment:
In other cases this is covered by [CA1805: Do not initialize unnecessarily](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1805). In those cases we want to instead update our [global configuration](https://github.com/rjmurillo/EffectiveCSharp.Analyzers/blob/main/src/EffectiveCSharp.Analyzers/Build/Config/General.globalconfig) to enable that analyzer. If there is a gap in that analyzer, then we should file an issue for triage and contribute a fix.

There is also some cases of this is covered in Roslynator RCS1129 ([code](https://github.com/dotnet/roslynator/blob/f945094a3b9adc64cac7bed03d587ce5a053fcdc/src/Analyzers/CSharp/Analysis/RemoveRedundantFieldInitializationAnalyzer.cs#L13), [docs](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1129/)).